spec/core/ics-002-client-semantics/README.md

```diff
@@ -196,95 +255,64 @@ A validity predicate is an opaque function defined by a client type to verify `H
 Using the validity predicate SHOULD be far more computationally efficient than replaying the full consensus algorithm
 for the given parent `Header` and the list of network messages.
 
-The validity predicate & client state update logic are combined into a single `checkValidityAndUpdateState` type, which is defined as
+The validity predicate is defined as:
 
 ```typescript
-type checkValidityAndUpdateState = (Header) => Void
+type VerifyClientMessage = (ClientMessage) => Void
 ```
 
-`checkValidityAndUpdateState` MUST throw an exception if the provided header was not valid.
-
-If the provided header was valid, the client MUST also mutate internal state to store
-now-finalised consensus roots and update any necessary signature authority tracking (e.g.
-changes to the validator set) for future calls to the validity predicate.
-
-Clients MAY have time-sensitive validity predicates, such that if no header is provided for a period of time
-(e.g. an unbonding period of three weeks) it will no longer be possible to update the client.
-In this case, a permissioned entity such as a chain governance system or trusted multi-signature MAY be allowed
-to intervene to unfreeze a frozen client & provide a new correct header.
+`VerifyClientMessage` MUST throw an exception if the provided ClientMessage was not valid.
 
 #### Misbehaviour predicate
 
-A misbehaviour predicate is an opaque function defined by a client type, used to check if data
+A misbehaviour predicate is an opaque function defined by a client type, used to check if a ClientMessage
 constitutes a violation of the consensus protocol. This might be two signed headers
 with different state roots but the same height, a signed header containing invalid
 state transitions, or other proof of malfeasance as defined by the consensus algorithm.
 
-The misbehaviour predicate & client state update logic are combined into a single `checkMisbehaviourAndUpdateState` type, which is defined as
+The misbehaviour predicate is defined as
 
 ```typescript
-type checkMisbehaviourAndUpdateState = (bytes) => Void
+type checkForMisbehaviour = (ClientMessage) => bool
```

Comment:
good point, `verifyClientMessage` is for authenticating the message is from the blockchain itself. In the tendermint case, this would be verifying the headers were signed by valid validator sets.

`checkForMisbehaviour` checks if the client message constitutes misbehaviour. This could be explicit, if the client message contains two headers check that they are conflicting.

Or implicit, if the clientmessage contains one header but it conflicts with a consensus state that already exists on chain.

`verifyHeaderAndUpdateState` used to verify the header, check for implicit misbehaviour, and then update state.

We split these up into explicit functions so light client developers did not miss implementing a critical piece.

See ADR for more details